scylla/src/transport/session.rs

```diff
@@ -1315,7 +1344,7 @@ impl Session {
         .await
     }
 
-    fn calculate_token(
+    pub fn calculate_token(
```

Comment:
I'm not sure if this is enough to resolve #468.

Reply:
Hmm looks like it's not, as this specifically asks for shard, and not only token (which would improve batching abilities).

Reply:
It's unclear for me how this should be done and likely would conflict with #491.

Reply:
Don't worry about conflicts with #491, it's a larger effort and we'll make sure that it works correctly with all previously added features. As for #468, our driver is internally capable of computing which shard owns a particular token (if all the information is properly fetched and available), take a look at this test case of a Sharder: https://github.com/scylladb/scylla-rust-driver/blob/fd06928929c0dd78a72c7aca494a998da1514a79/scylla/src/routing.rs#L144-L166

So if you go deep enough, it's possible to expose this sharding information publicly.

Reply:
Sharder appears to be a property of a Node, so it looks like I'd have to go through building the query plan, getting the corresponding node (what if there are several nodes output by the plan? just take the first one?) then get the sharder for the node and eventually return something like a (node, shard) pair, but I don't know what we could use as node identifier, in order to allow users to group by that.

It looks like providing an interface for computing the token is a good first step still, as it will allow grouping at least by partition if not by shard, and exposing the shard interface can be done in a later PR.
As it's unlikely I'll have time for implementing the shard interface anytime soon, I'd suggest moving forward with this without this additional feature.